src/kixi/datastore/metadatastore/inmemory.clj

```diff
@@ -3,19 +3,49 @@
             [com.stuartsierra.component :as component]
             [kixi.datastore.metadatastore
              :refer [MetaDataStore] :as ms]
-            [kixi.datastore.communications
-             :refer [attach-sink-processor]]
+            [kixi.comms :as c]
             [taoensso.timbre :as timbre :refer [error info infof]]))
 
-(defn update-metadata-processor
-  [data]
-  (fn [metadata]
+(s/fdef update-metadata-processor
+        :args (s/cat :data #(instance? clojure.lang.IAtom %)
+                     :update-req ::ms/file-metadata-updated))
+
+(defmulti update-metadata-processor
+  (fn [data update-event]
+    (::ms/file-metadata-update-type update-event)))
+
+(defmethod update-metadata-processor ::ms/file-metadata-created
+  [data update-event]
+  (let [metadata (::ms/file-metadata update-event)]
     (info "Update: " metadata)
     (swap! data
            #(update % (::ms/id metadata)
                     (fn [current-metadata]
                       (merge current-metadata
-                             metadata))))))
+                             metadata))))
+    nil))
+
+
+(defmethod update-metadata-processor ::ms/file-metadata-structural-validation-checked
+  [data update-event]
+  (info "Update: " update-event)
+  (swap! data
+         #(update % (::ms/id update-event)
+                  (fn [current-metadata]
+                    (assoc (or current-metadata {})
+                           ::ms/structural-validation (::ms/structural-validation update-event)))))
+  nil)
```

Comment:
This is a nasty way of not emitting an event from an event consumer. It's a side effect of the comms module not really knowing what to expect back from a handler fn.